sdk/src/Vyper.ts

```diff
@@ -8,17 +8,39 @@ import { LastUpdate } from "./LastUpdate";
 import { ReserveFairValue } from "./ReserveFairValue";
 import { TrancheData } from "./TrancheData";
 import { TrancheFairValue } from "./TrancheFairValue";
+import { RedeemLogicLending } from "../../target/types/redeem_logic_lending";
+import idlRedeemLogicLending from "../../target/idl/redeem_logic_lending.json";
+import { RedeemLogicState } from "./RedeemLogicState";
+import idlRateMock from "../../target/idl/rate_mock.json";
+import { RateMock } from "../../target/types/rate_mock";
+import { RateState } from "./RateMockState";
+
 export class Vyper {
 
     program: anchor.Program<VyperCore>;
     provider: anchor.AnchorProvider;
     trancheId: PublicKey;
+    redeemLendingProgram: anchor.Program<RedeemLogicLending>;
+    redeemLendingStateId: PublicKey;
+    rateMockProgram: anchor.Program<RateMock>;
+    rateMockStateId: PublicKey;
 
-    static create(provider: anchor.AnchorProvider, vyperCoreId: PublicKey): Vyper {
+    static create(provider: anchor.AnchorProvider, vyperCoreId: PublicKey, redeemLendingId?: PublicKey, rateMockId?: PublicKey): Vyper {
```

Comment:
we need to **decouple** the logic from Vyper and the plugins, we can build an interface for both the plugins families and here pass those interfaces. this will surely help in the future when we'll have multiple plugins

Reply:
Done, made interfaces for each plugin and have them in vyper core separately.